src/extension/wps/wps-core/src/main/java/org/geoserver/wps/gs/SpatioTemporalZonalStatistics.java

```diff
@@ -0,0 +1,583 @@
+import org.geotools.api.referencing.operation.TransformException;
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.grid.io.AbstractGridFormat;
+import org.geotools.coverage.grid.io.GridCoverage2DReader;
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.data.simple.SimpleFeatureIterator;
+import org.geotools.feature.collection.DecoratingSimpleFeatureCollection;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.geometry.jts.JTS;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.process.ProcessException;
+import org.geotools.process.factory.DescribeParameter;
+import org.geotools.process.factory.DescribeProcess;
+import org.geotools.process.factory.DescribeResult;
+import org.geotools.process.raster.CoverageUtilities;
+import org.geotools.referencing.CRS;
+import org.geotools.util.DateRange;
+import org.geotools.util.logging.Logging;
+import org.jaitools.media.jai.zonalstats.ZonalStats;
+import org.jaitools.media.jai.zonalstats.ZonalStatsDescriptor;
+import org.jaitools.media.jai.zonalstats.ZonalStatsOpImage;
+import org.jaitools.numeric.Range;
+import org.jaitools.numeric.Statistic;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.Geometry;
+
+/**
+ * A GeoServer process for computing spatio-temporal zonal statistics on multi-temporal raster coverages.
+ *
+ * <p>This process allows computing statistical aggregations (such as min, max, sum, mean, median) across multiple time
+ * steps for specific zone polygon features. It supports time specification through either a comma-separated list of
+ * timestamps or a time range.
+ *
+ * <p>When specifying a time range, the process will extract the time domain from the underlying coverage limiting the
+ * results to a max of 1000 time entries (configurable via the "spatio.temporal.max.entries" system property).
+ */
+@DescribeProcess(
+        title = "SpatioTemporal Zonal statistics",
+        description = "Compute aggregated zonal statistics on a multi-temporal coverage")
+public class SpatioTemporalZonalStatistics implements GeoServerProcess {
+
+    private static final Integer[] BAND_STAT = {0};
+
+    public static final int MAX_TIME_ENTRIES =
+            Integer.parseInt(System.getProperty("spatio.temporal.max.entries", "1000"));
+
+    private static final EnumSet<Statistic> ALLOWED_STATS =
+            EnumSet.of(Statistic.MIN, Statistic.MAX, Statistic.SUM, Statistic.MEAN, Statistic.MEDIAN);
+
+    static final Logger LOGGER = Logging.getLogger(SpatioTemporalZonalStatistics.class);
+
+    private Catalog catalog;
+
+    public SpatioTemporalZonalStatistics(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
+    @DescribeResult(
+            name = "result",
+            description = "A feature collection containing aggregated statistics on each zone",
```

Comment:
I would expand a bit, this is most of what they user will see. If  it does not fit well here in the code, you can probably create a separate constant and refer it from here.